os/include/tinyara/mm/mm.h

```diff
@@ -791,6 +791,19 @@ void *zalloc_at(int heap_index, size_t size);
 #define zalloc_at(heap_index, size)              zalloc(size)
 #endif
 
+/**
+ * @brief Allocate to specified user heap.
+ * @details @b #include <tinyara/mm/mm.h>\n
+ *   This is a simple wrapper for the mm_malloc() function.
+ *   This function is used to allocate the memory from kernel side
+ *   to specified user heap.
+ */
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+void *umm_heap_malloc(struct mm_heap_s *heap, size_t size, mmaddress_t caller_retaddr);
+#else
+void *umm_heap_malloc(struct mm_heap_s *heap, size_t size);
+#endif
```

Comment:
Because this is a public API, it should not cause two different usages on the user side.
This should be just
```
void *umm_heap_malloc(struct mm_heap_s *heap, size_t size);
```
`mmaddress_t caller_retaddr` should be concerned inside of the new function.
Then it will be the same as other public API.
Additionally the purpose of the `caller_retaddr` is figuring out the point where calls memory allocation.
If we do as commit does, it will return caller of elf_loaddtors, not elf_loaddtors.
It should be go into the new function.

Reply:
Modified. Thanks.